src/cloud/config_manager/controllers/server.go

```diff
@@ -217,12 +219,37 @@ func (s *Server) GetConfigForVizier(ctx context.Context,
 	}
 	AddDefaultTableStoreSize(tmplValues.PEMMemoryRequest, tmplValues.CustomPEMFlags)
 
-	// Next we inject any feature flags that we want to set for this org.
-	orgID, err := s.getOrgIDForDeployKey(tmplValues.DeployKey)
+	// Attempt to get the org ID from DeployKey, otherwise from the Vizier.
+	var orgID uuid.UUID
+	orgID, err = s.getOrgIDForDeployKey(tmplValues.DeployKey)
 	if err != nil || orgID == uuid.Nil {
-		log.WithError(err).Error("Error getting org ID from deploy key, skipping feature flag logic")
-	} else {
+		log.WithError(err).Error("Error getting org ID from deploy key")
+	}
+	if orgID == uuid.Nil && in.VizierID != "" {
+		// Generate token to use to get the org information from Vizier.
+		claims := srvutils.GenerateJWTForService("vzmgr Service", viper.GetString("domain_name"))
+		token, err := srvutils.SignJWTClaims(claims, viper.GetString("jwt_signing_key"))
+		if err != nil {
+			return nil, err
+		}
+
+		ctx = metadata.AppendToOutgoingContext(context.Background(), "authorization",
+			fmt.Sprintf("bearer %s", token))
+
+		resp, err := s.vzmgrClient.GetOrgFromVizier(ctx, utils.ProtoFromUUIDStrOrNil(in.VizierID))
+		if err != nil {
+			log.WithError(err).Error("Error getting the org ID from Vizier")
+		} else {
+			orgID = utils.UUIDFromProtoOrNil(resp.OrgID)
+		}
```

Comment:
IMO it's worth pulling this out into `getOrgIDForVizier` similar to how `getOrgIDForDeployKey` is a helper func.